imports/plugins/included/product-detail-simple/client/components/productDetail.js

```diff
@@ -70,7 +92,122 @@ class ProductDetail extends Component {
         </Toolbar>
       );
     }
+    return null;
+  }
+  switchDigital(e) {
+    this.setState({digital: e.target.checked});
+    const productId = this.props.product._id;
+    const checked = e.target.checked;
+    this.props.onProductFieldChange(productId, "isDigital", checked);
+    this.props.changeParentIsDigitalState(e.target.checked);
+    if (!this.state.digital) {
+      this.setState({cartQuantityType: "hidden"});
+    } else {
+      this.setState({cartQuantityType: "number"});
+    }
+  }
+
+  renderInputToggle() {
+    if (this.props.hasAdminPermission) {
+      return (
+        <div className="switch-middle">
+          <b className="switch-text">Physical Product</b>
+            <label className="switch">
+              <input id="digital" type="checkbox" defaultChecked={this.state.digital} onChange={this.switchDigital}/>
+              <div className="slider round"/>
+            </label>
+          <b className="switch-text">Digital Product</b>
+        </div>
+      );
+    }
+
+    return null;
+  }
+
+  handleCategoryStateChange(e) {
+    const value = e.target.value;
```

Comment:
No need of instantiating a variable for this since we're using it just one place and there's no need to compute the value